### PR TITLE
Stay connected : save session info after successful login

### PR DESCRIPTION
**Pull request by sujeebant**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -160,9 +162,9 @@ class Masq {
     })
   }
 
-  async _isRegistered () {
+  async _isRegistered (userId) {
     // is registered if the userId is the id of a known db
-    return !this.userId || utils.dbExists(this.userId)
+    return !userId || utils.dbExists(userId)
```
Comment (G-Ray) [inline src/index.js]: :no_entry:  If we call `_isRegistered(false)` or `_isRegistered(false)` it will return `true`

Comment (G-Ray) [reply src/index.js]: and if userId is defined, it will return true

File: test/masq.test.js [hunk 2]
```diff
@@ -555,3 +574,50 @@ test('should set a watcher', async (done) => {
   masq.watch('/hello', onChange)
```
Comment (G-Ray) [inline test/masq.test.js]: The test misses a check on the values passed to the cb, can you add the check please ?

File: src/index.js (hunk 1 above)
Comment (sujeebant) [reply src/index.js]: fixed

File: src/index.js [hunk 3]
```diff
@@ -87,14 +89,21 @@ class Masq {
...
+  _storeSessionInfo (stayConnected, userId) {
+    if (stayConnected) {
+      window.localStorage.setItem('userId', userId)
+    }
+    window.sessionStorage.setItem('userId', userId)
   }
 
   _loadSessionInfo () {
     this.userId = window.localStorage.getItem('userId')
+    if (this.userId) {
```
Comment (G-Ray) [inline src/index.js]: ok

File: src/index.js [hunk 4]
```diff
@@ -160,9 +169,12 @@ class Masq {
...
 
-  async _isRegistered () {
-    // is registered if the userId is the id of a known db
-    return !this.userId || utils.dbExists(this.userId)
+  async _isRegistered (userId) {
+    // is registered if :
+    // - this.userId is set (loaded in Masq constructor)
+    // - this.userId is the same as userId
+    // - there is a db for this userId
+    return (this.userId && this.userId === userId && utils.dbExists(userId))
```
Comment (G-Ray) [inline src/index.js]: In this case, why take a param named `userId` ? the result of this function does not depend on it it seems. It only depends on `this.userId`

Comment (sujeebant) [reply src/index.js]: We do check that the input userId is equal to attrubute this.userId

File: src/index.js [hunk 5]
```diff
@@ -160,9 +162,12 @@ class Masq {
...
 
-  async _isRegistered () {
-    // is registered if the userId is the id of a known db
-    return !this.userId || utils.dbExists(this.userId)
+  async _isRegistered (userId) {
+    // is registered if :
+    // - this.userId is set (loaded in Masq constructor)
+    // - this.userId is the same as userId
+    // - there is a db for this userId
+    return (this.userId && this.userId !== userId && utils.dbExists(userId))
```
[the remark this answers is not in the document]
Comment (sujeebant) [reply src/index.js]: fixed

File: src/index.js (hunk 4 above)
Comment (sujeebant) [reply src/index.js]: We could only check if dbExists(userId)

**Pull request merged**